app/feature/insurances/src/main/kotlin/com/hedvig/android/feature/insurances/insurance/presentation/InsurancePresenter.kt

```diff
@@ -145,23 +68,96 @@ internal data class InsuranceUiState(
   }
 }
 
+internal class InsurancePresenter(
+  private val getInsuranceContractsUseCase: GetInsuranceContractsUseCase,
+  private val getCrossSellsUseCase: GetCrossSellsUseCase,
+  private val crossSellCardNotificationBadgeService: CrossSellCardNotificationBadgeService,
+) : MoleculePresenter<InsuranceScreenEvent, InsuranceUiState> {
+  @Composable
+  override fun MoleculePresenterScope<InsuranceScreenEvent>.present(
+    seed: InsuranceUiState,
+  ): InsuranceUiState {
+    var insuranceData by remember {
+      mutableStateOf<InsuranceData>(
+        InsuranceData(
+          insuranceCards = seed.insuranceCards,
+          crossSells = seed.crossSells,
+          quantityOfCancelledInsurances = seed.quantityOfCancelledInsurances,
+        ),
+      )
+    }
+    var isLoading by remember { mutableStateOf(true) }
+    var didFailToLoad by remember { mutableStateOf(false) }
+    val retryChannel = remember { RetryChannel() }
+
+    val showNotificationBadge by crossSellCardNotificationBadgeService
+      .showNotification()
+      .collectAsState(seed.showNotificationBadge)
```

Comment:
I realized before we were reacting to the status of the notification only on each time we refresh, which is not what we were really looking for here.
Since the state comes from inside DataStore as a `Flow<Boolean>`, we can directly listen to it and turn it into state with a normal `collectAsState(initialValue = ...)`.
No we get the reactivity for free, if DataStore ever updates, this auto-updates and the presenter will emit a new version of the UI state